mailpoet/mailpoet
# Forms: per form cookies and custom expiry date [MAILPOET-3059]

With this PR each form (of types popup, slide In, or Fixed bar) will have its own cookie.

It is possible to specify the expiry date in the form editor from the following values:
  - Always (session cookie)
  - 1 day
  - 3 days
  - 7 days (default)
  - 14 days
  - 30 days
  - 60 days
  - 90 days

Forms will not be displayed if the user is logged in and subscribed to any of the segments in the form

After subscription, the cookie is set with an expiration time of 182 days.

When this PR will be released, forms will be displayed again for visitors that had the previous cookie set up.

How to test:
* Create or edit a form, check that you can see the following dropdown when placement is `popup`, `slide in`, or `fixed bar`
![ShowFormEvery](https://user-images.githubusercontent.com/8002881/153954656-c47aa22c-73e0-4ec8-9178-3a6405258285.png)

* Select one of the values on the dropdown, one list you are not subscribed to, any other required data and save the form.

* Log out, visit the home page, check that the form is displayed, subscribe. Check the cookie is set with 182 days expiration time.
* Remove the cookie, log in. Check the form is not displayed.
* Log out, visit the home page, check the form is displayed, dismiss the form, check the cookie is set with the expiration time selected in the editor.


Other things to test:
- Cookie with 182 days expiration time is also set on redirection after submission, with or without Captcha
- Cookie is not set when previewing the form
- Selecting `always` on the drop-down value will set a session cookie
- Cookie is removed when saving the form

[MAILPOET-3059]

[MAILPOET-3059]: https://mailpoet.atlassian.net/browse/MAILPOET-3059?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
